### PR TITLE
Decouple HorovodPytorchEstimator from ray sgd

### PR DESCRIPTION
**Pull request by yangw1234**

### CODE REVIEW

**Comment by yangw1234** [comment]
http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/3757/ jenkins passed (example download error is irrelevant)

File: pyzoo/zoo/orca/learn/pytorch/training_operator.py [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Some portions of this file Copyright 2017 The Ray Authors.
+# and licensed under the Apache License, Version 2.0
+#
+
+# This file is adapted from
+# https://github.com/ray-project/ray/blob/master/python/ray/util/sgd/torch/training_operator.py
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/pytorch/training_operator.py:21]: add appropriate copyright as in https://github.com/ray-project/ray/blob/master/src/ray/core_worker/common.cc

File: pyzoo/zoo/orca/learn/pytorch/torch_runner.py [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Some portions of this file Copyright 2017 The Ray Authors.
+# and licensed under the Apache License, Version 2.0
+#
+
+# This file is adapted from
+# https://github.com/ray-project/ray/blob/master/python/ray/util/sgd/torch/torch_runner.py
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/pytorch/torch_runner.py:21]: add appropriate copyright as in https://github.com/ray-project/ray/blob/master/src/ray/core_worker/common.cc

File: pyzoo/zoo/orca/learn/pytorch/utils.py [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Some portions of this file Copyright 2017 The Ray Authors.
+# and licensed under the Apache License, Version 2.0
+#
+
+# This file is adapted from
+# https://github.com/ray-project/ray/blob/master/python/ray/util/sgd/utils.py
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/pytorch/utils.py:21]: add appropriate copyright as in https://github.com/ray-project/ray/blob/master/src/ray/core_worker/common.cc

File: pyzoo/zoo/orca/learn/pytorch/pytorch_horovod_estimator.py [hunk 4]
```diff
@@ -69,28 +70,11 @@ def _setup_training(self):
             raise ValueError("only support one optimizer for now")
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/pytorch/pytorch_horovod_estimator.py:70]: remove cuda

File: pyzoo/zoo/orca/learn/pytorch/pytorch_horovod_estimator.py [hunk 5]
```diff
@@ -108,21 +92,83 @@ def with_sampler(loader):
...
+        map it onto the actor's specific device.
+
+        From: github.com/pytorch/pytorch/issues/10622#issuecomment-474733769
+        """
+        _buffer = io.BytesIO(byte_obj)
+        to_gpu = self.use_gpu and torch.cuda.is_available()
+        state_dict = torch.load(
+            _buffer,
+            map_location=("cpu" if not to_gpu else
+                          lambda storage, loc: storage.cuda()))
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/pytorch/pytorch_horovod_estimator.py:136]: remove cuda

**Pull request merged**